test/integration_test.py

```diff
@@ -310,9 +310,13 @@ def _check_terra_bdbag(self, response: bytes):
             data_path = os.path.join(os.path.dirname(first(zip_fh.namelist())), 'data')
             file_path = os.path.join(data_path, 'participants.tsv')
             with zip_fh.open(file_path) as file:
-                self.__check_manifest(file, 'bundle_uuid')
+                rows = self.__check_manifest(file, 'bundle_uuid')
+                for row in rows:
```

Comment:
I don't like modal args in general but I also didn't like how in your verison the control flow forks for bdbag and compact and then rejoins and then forks again.